tests/python/unittest/test_custom_datatypes_change_dtype.py

```diff
@@ -0,0 +1,553 @@
+        with tvm.transform.PassContext(config={"tir.disable_vectorize": True}):
+            maybe_correct = ex.evaluate()(x_converted)
+            maybe_correct_converted = convert_ndarray(src_dtype, maybe_correct)
+        np.testing.assert_allclose(maybe_correct_converted.asnumpy(),
+                                   correct.asnumpy(),
+                                   rtol=rtol,
+                                   atol=atol)
+        # print(maybe_correct_converted)
+        # print(correct)
+
+    for op in [
+            relay.nn.softmax,
+            tvm.relay.log,
+            tvm.relay.exp,
+            tvm.relay.sqrt,
+            tvm.relay.rsqrt,
+            tvm.relay.sigmoid,
+            tvm.relay.tanh,
+            relay.nn.relu,
+    ]:
+        check_unary_op(op, src_dtype, dst_dtype)
+
+    def check_binary_op(opfunc, src_dtype, dst_dtype):
+        t1 = relay.TensorType((5, 10, 5), src_dtype)
+        t2 = relay.TensorType((5, ), src_dtype)
+        x = relay.var("x", t1)
+        y = relay.var("y", t2)
+        z = opfunc(x, y)
+        x_data = np.random.rand(5, 10, 5).astype(t1.dtype)
+        y_data = np.random.rand(5).astype(t2.dtype)
+        module = tvm.IRModule.from_expr(relay.Function([x, y], z))
+
+        ex = relay.create_executor("graph", mod=module)
+
+        correct = ex.evaluate()(x_data, y_data)
+
+        module, _ = change_dtype(src_dtype, dst_dtype, module, [])
+        ex = relay.create_executor("graph", mod=module)
+
+        x_converted = convert_ndarray(dst_dtype, x_data)
+        y_converted = convert_ndarray(dst_dtype, y_data)
+
+        with tvm.transform.PassContext(config={"tir.disable_vectorize": True}):
+            maybe_correct = ex.evaluate()(x_converted, y_converted)
+            maybe_correct_converted = convert_ndarray(src_dtype, maybe_correct)
+        np.testing.assert_allclose(correct.asnumpy(),
+                                   maybe_correct_converted.asnumpy(),
+                                   rtol=rtol,
+                                   atol=atol)
+
+    for op in [
+            relay.add,
+            relay.subtract,
+            relay.divide,
+            relay.multiply,
+    ]:
+        check_binary_op(op, src_dtype, dst_dtype)
+
+
+def run_model(get_workload,
+              input_shape,
+              src_dtype,
+              dst_dtype,
+              num_classes,
+              rtol=0.0001,
+              atol=0.0001):
+    module, params = get_workload(image_shape=input_shape,
+                                  num_classes=num_classes)
+
+    # Convert the input into the correct format.
+    input = tvm.nd.array(np.random.rand(*input_shape).astype(src_dtype))
+
+    ex = relay.create_executor('graph', mod=module)
+    correct = ex.evaluate()(input, **params)
+
+    # Simplifying inference is essential right now, as batch norms (which get
+    # removed) are broken with custom datatypes.
+    #expr = relay.ir_pass.simplify_inference(expr)
```

Comment:
@gussmith23 why is this here?

Reply:
For a while (not sure if still true -- we should set up a test for it) batch norms were broken with custom datatypes. It was a very odd bug that I didn't end up figuring out. It seemed to go pretty deep, and be related to the code being generated. However, the batch norm operator can generally be optimized out when you're running a network for inference (and not training), which is what `simplify_inference` did. So we'd run `simplify_inference` to remove the batch norms and things would be hunky-dory.

Reply:
@gussmith23 Still need to document this somewhere.